core/src/main/java/com/klaytn/caver/transaction/type/LegacyTransaction.java

```diff
@@ -270,4 +271,38 @@ public String getInput() {
     public String getValue() {
         return value;
     }
+
+    public void setTo(String to) {
+        // "to" field in LegacyTransaction allows null
+        if(to == null || to.isEmpty()) {
+            to = "0x";
+        }
+
+        if(!to.equals("0x") && !Utils.isAddress(to)) {
+            throw new IllegalArgumentException("Invalid address. : " + to);
+        }
+        this.to = to;
+    }
+
+    public void setInput(String input) {
+        if(input == null) {
+            throw new IllegalArgumentException("input is missing");
+        }
+
+        if(!input.equals("0x") && !Utils.isHex(input)) {
+            throw new IllegalArgumentException("Invalid input.");
+        }
+        this.input = input;
+    }
+
+    public void setValue(String value) {
+        if(value == null) {
+            throw new IllegalArgumentException("value is missing");
+        }
+
+        if(!Utils.isHex(value)) {
```

Comment:
How about adding this validation or other(isNumber??) to other setter? for example setGasPrice or setNonce.